core/src/main/java/apoc/trigger/TriggerNewProcedures.java

```diff
@@ -0,0 +1,129 @@
+package apoc.trigger;
+
+import apoc.SystemPropertyKeys;
+import apoc.util.Util;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.api.procedure.SystemProcedure;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.Admin;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Mode;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.neo4j.configuration.GraphDatabaseSettings.SYSTEM_DATABASE_NAME;
+
+
+public class TriggerNewProcedures {
+    // public for testing purpose
+    public static final String TRIGGER_NOT_ROUTED_ERROR = "The procedure should be routed and executed against a writer database";
+    public static final String TRIGGER_BAD_TARGET_ERROR = "Triggers can only be installed on user databases.";
+
+    @Context public GraphDatabaseAPI db;
+    
+    @Context public Log log;
+    
+    @Context public Transaction tx;
+
+    private void checkInSystemWriter() {
+        TriggerHandlerNewProcedures.checkEnabled();
+        // routing check
```

Comment:
I'm not sure how useful this comment is, especially because the name of this `checkInSystemWriter` function is fairly descriptive about what it does